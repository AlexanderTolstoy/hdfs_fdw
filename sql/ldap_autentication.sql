/*---------------------------------------------------------------------------------------------------------------------
 *
 * ldap_autentication.sql
 * 		Foreign-data wrapper for remote Hadoop servers
 *              To verify the LDAP Authentication feature is working successfully.
 *
 * Portions Copyright (c) 2012-2014, PostgreSQL Global Development Group
 *
 * Portions Copyright (c) 2004-2014, EnterpriseDB Corporation.
 *
 * IDENTIFICATION
 * 		ldap_autentication.sql
 *
 *---------------------------------------------------------------------------------------------------------------------
 */

-- Connection Settings.

\set HIVE_SERVER         `echo \'"$HIVE_SERVER"\'`
\set HIVE_CLIENT_TYPE           '\'hiveserver2\''
\set HIVE_PORT           `echo \'"$HIVE_PORT"\'`
\set HIVE_USER           `echo \'"$HIVE_USER"\'`
\set HIVE_PASSWORD       `echo \'"$HIVE_PASSWORD"\'`

-- Create the database.

CREATE DATABASE fdw_regression;

\c fdw_regression postgres

-- Set the Date Style

SET datestyle TO SQL,DMY;

-- Set the Search Path to PUBLIC Schema

SET search_path TO public;

-- Create Hadoop FDW Extension.

CREATE EXTENSION hdfs_fdw;

-- Create Hadoop FDW Server.

CREATE SERVER hdfs_server FOREIGN DATA WRAPPER hdfs_fdw OPTIONS(host :HIVE_SERVER, port :HIVE_PORT, client_type :HIVE_CLIENT_TYPE);

--**************************************************************************************************
-- To verify that error message displayed when invalid LDAP User is used
--**************************************************************************************************

-- Create postgres USER MAPPING with incorrect LDAP User.

CREATE USER MAPPING FOR postgres SERVER hdfs_server OPTIONS (username 'wronguser', password 'edb'); 

-- Create Foreign Tables.

CREATE FOREIGN TABLE dept_dt_mp1 
(
	deptno INTEGER, 
	dname VARCHAR(14), 
	loc VARCHAR(13) ) 
SERVER hdfs_server OPTIONS (dbname 'fdw_db', table_name 'dept_dt_mp');

-- Error message will be displayed as the wrong LDAP User is mentioned in User Mapping.

SELECT * FROM dept_dt_mp1;

-- Drop user mapping and create correct user mapping.

DROP USER MAPPING FOR postgres SERVER hdfs_server;

CREATE USER MAPPING FOR postgres SERVER hdfs_server OPTIONS (username :HIVE_USER, password :HIVE_PASSWORD);

-- Data will be displayed.

SELECT * FROM dept_dt_mp1;

--**************************************************************************************************
-- To verify that error message displayed when invalid LDAP User password is used
--**************************************************************************************************

-- Drop user mapping.

DROP USER MAPPING FOR postgres SERVER hdfs_server;

-- Create postgres USER MAPPING with incorrect LDAP User password.

CREATE USER MAPPING FOR postgres SERVER hdfs_server OPTIONS (username 'kzeeshan', password 'wrong');

-- Error message will be displayed as the wrong LDAP User password is mentioned in User Mapping.

SELECT * FROM dept_dt_mp1;

-- Drop user mapping and create correct user mapping.

DROP USER MAPPING FOR postgres SERVER hdfs_server;

CREATE USER MAPPING FOR postgres SERVER hdfs_server OPTIONS (username :HIVE_USER, password :HIVE_PASSWORD);

-- Data will be displayed.

SELECT * FROM dept_dt_mp1;

--**************************************************************************************************
-- To verify that error message displayed when invalid LDAP User and invalid password is used
--**************************************************************************************************

-- Drop user mapping.

DROP USER MAPPING FOR postgres SERVER hdfs_server;

-- Create postgres USER MAPPING with invalid LDAP User and invalid password.

CREATE USER MAPPING FOR postgres SERVER hdfs_server OPTIONS (username 'wrong', password 'wrong');

-- Error message will be displayed as the wrong LDAP User password is mentioned in User Mapping.

SELECT * FROM dept_dt_mp1;

-- Drop user mapping and create correct user mapping.

DROP USER MAPPING FOR postgres SERVER hdfs_server;

CREATE USER MAPPING FOR postgres SERVER hdfs_server OPTIONS (username :HIVE_USER, password :HIVE_PASSWORD);

-- Data will be displayed.

SELECT * FROM dept_dt_mp1;



--Cleanup

DROP FOREIGN TABLE dept_dt_mp1;

DROP EXTENSION hdfs_fdw CASCADE;


\c postgres postgres

DROP DATABASE fdw_regression;
