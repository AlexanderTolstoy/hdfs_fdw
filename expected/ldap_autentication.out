/*---------------------------------------------------------------------------------------------------------------------
 *
 * ldap_autentication.sql
 * 		Foreign-data wrapper for remote Hadoop servers
 *              To verify the LDAP Authentication feature is working successfully.
 *
 * Portions Copyright (c) 2012-2014, PostgreSQL Global Development Group
 *
 * Portions Copyright (c) 2004-2014, EnterpriseDB Corporation.
 *
 * IDENTIFICATION
 * 		ldap_autentication.sql
 *
 *---------------------------------------------------------------------------------------------------------------------
 */
-- Connection Settings.
\set HIVE_SERVER         `echo \'"$HIVE_SERVER"\'`
\set HIVE_CLIENT_TYPE           '\'hiveserver2\''
\set HIVE_PORT           `echo \'"$HIVE_PORT"\'`
\set HIVE_USER           `echo \'"$HIVE_USER"\'`
\set HIVE_PASSWORD       `echo \'"$HIVE_PASSWORD"\'`
-- Create the database.
CREATE DATABASE fdw_regression;
\c fdw_regression postgres
-- Set the Date Style
SET datestyle TO SQL,DMY;
-- Set the Search Path to PUBLIC Schema
SET search_path TO public;
-- Create Hadoop FDW Extension.
CREATE EXTENSION hdfs_fdw;
-- Create Hadoop FDW Server.
CREATE SERVER hdfs_server FOREIGN DATA WRAPPER hdfs_fdw OPTIONS(host :HIVE_SERVER, port :HIVE_PORT, client_type :HIVE_CLIENT_TYPE);
--**************************************************************************************************
-- To verify that error message displayed when invalid LDAP User is used
--**************************************************************************************************
-- Create postgres USER MAPPING with incorrect LDAP User.
CREATE USER MAPPING FOR postgres SERVER hdfs_server OPTIONS (username 'wronguser', password 'edb'); 
-- Create Foreign Tables.
CREATE FOREIGN TABLE dept_dt_mp1 
(
	deptno INTEGER, 
	dname VARCHAR(14), 
	loc VARCHAR(13) ) 
SERVER hdfs_server OPTIONS (dbname 'fdw_db', table_name 'dept_dt_mp');
-- Error message will be displayed as the wrong LDAP User is mentioned in User Mapping.
SELECT * FROM dept_dt_mp1;
ERROR:  failed to initialize the HDFS connection object (ERROR : Could not connect to jdbc:hive2://127.0.0.1:10000/fdw_db within 300000 seconds)
-- Drop user mapping and create correct user mapping.
DROP USER MAPPING FOR postgres SERVER hdfs_server;
CREATE USER MAPPING FOR postgres SERVER hdfs_server OPTIONS (username :HIVE_USER, password :HIVE_PASSWORD);
-- Data will be displayed.
SELECT * FROM dept_dt_mp1;
 deptno |   dname    |      loc      
--------+------------+---------------
     10 | ACCOUNTING | NEW YORK     
     20 | RESEARCH   | DALLAS       
     30 | SALES      | CHICAGO      
     40 | OPERATIONS | BOSTON       
(4 rows)

--**************************************************************************************************
-- To verify that error message displayed when invalid LDAP User password is used
--**************************************************************************************************
-- Drop user mapping.
DROP USER MAPPING FOR postgres SERVER hdfs_server;
-- Create postgres USER MAPPING with incorrect LDAP User password.
CREATE USER MAPPING FOR postgres SERVER hdfs_server OPTIONS (username 'kzeeshan', password 'wrong');
-- Error message will be displayed as the wrong LDAP User password is mentioned in User Mapping.
SELECT * FROM dept_dt_mp1;
ERROR:  failed to initialize the HDFS connection object (ERROR : Could not connect to jdbc:hive2://127.0.0.1:10000/fdw_db within 300000 seconds)
-- Drop user mapping and create correct user mapping.
DROP USER MAPPING FOR postgres SERVER hdfs_server;
CREATE USER MAPPING FOR postgres SERVER hdfs_server OPTIONS (username :HIVE_USER, password :HIVE_PASSWORD);
-- Data will be displayed.
SELECT * FROM dept_dt_mp1;
 deptno |   dname    |      loc      
--------+------------+---------------
     10 | ACCOUNTING | NEW YORK     
     20 | RESEARCH   | DALLAS       
     30 | SALES      | CHICAGO      
     40 | OPERATIONS | BOSTON       
(4 rows)

--**************************************************************************************************
-- To verify that error message displayed when invalid LDAP User and invalid password is used
--**************************************************************************************************
-- Drop user mapping.
DROP USER MAPPING FOR postgres SERVER hdfs_server;
-- Create postgres USER MAPPING with invalid LDAP User and invalid password.
CREATE USER MAPPING FOR postgres SERVER hdfs_server OPTIONS (username 'wrong', password 'wrong');
-- Error message will be displayed as the wrong LDAP User password is mentioned in User Mapping.
SELECT * FROM dept_dt_mp1;
ERROR:  failed to initialize the HDFS connection object (ERROR : Could not connect to jdbc:hive2://127.0.0.1:10000/fdw_db within 300000 seconds)
-- Drop user mapping and create correct user mapping.
DROP USER MAPPING FOR postgres SERVER hdfs_server;
CREATE USER MAPPING FOR postgres SERVER hdfs_server OPTIONS (username :HIVE_USER, password :HIVE_PASSWORD);
-- Data will be displayed.
SELECT * FROM dept_dt_mp1;
 deptno |   dname    |      loc      
--------+------------+---------------
     10 | ACCOUNTING | NEW YORK     
     20 | RESEARCH   | DALLAS       
     30 | SALES      | CHICAGO      
     40 | OPERATIONS | BOSTON       
(4 rows)

--Cleanup
DROP FOREIGN TABLE dept_dt_mp1;
DROP EXTENSION hdfs_fdw CASCADE;
NOTICE:  drop cascades to 2 other objects
DETAIL:  drop cascades to server hdfs_server
drop cascades to user mapping for postgres on server hdfs_server
\c postgres postgres
DROP DATABASE fdw_regression;
